### PR TITLE
Add hook for reacting to signal setters

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

**Review by dylhunn — APPROVED** [review]

**Comment by chrisrocco** [comment]
Thanks!

File: packages/core/src/signals/src/signal.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@ import {createSignalFromFunction, defaultEquals, Signal, ValueEqualityFn} from '
 import {throwInvalidWriteToSignalError} from './errors';
 import {ReactiveNode} from './graph';
 
+/**
+ * If set, called after `WritableSignal`s are updated.
+ *
+ * This hook can be used to achieve various effects, such as running effects synchronously as part
+ * of setting a signal.
+ */
+let postSignalSetFn: (() => void)|null = null;
```
Comment (jelbourn) [inline packages/core/src/signals/src/signal.ts:19]: optional / subjective: 
```suggestion
let afterSetCallback: (() => void)|null = null;
```

**Comment by alxhub** [comment]
Caretaker: TAP is green with the exception of a pre-existing failure.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit a6fc8b3cee93469cb8f650170cf9056d824ea3a0.

**Pull request closed without merging**